Use numbers for line and column in links to org headers, add tests

matsl opened the pull request:
# What

Use numbers for implicit links to org headers.

# Why

~Links were created with pattern file#header:L{number}:C{number} which is not identified as a link? Is is a bug? I'm surprised it passed our testing in that case but it seems I did not add any tests for this functionality.  But now it is there.~ My bad. It is an optional feature but then we could use the more standard link without the L and C. Any reason not to do that?

I noticed this while looking at #491 to figure out if link to string match would be possible to use easily.

I also added tests to document the changed behavior. A link to a header now looks file:#header:{number}:{number}.

# Note

Is there any direct way now to create a link to string match? Was thinking if we could prefer that link type if the region is active in the file to link to? Or would that be to subtle?

rswgnu commented:
Yes, there must be a number following the L and C or the link is invalid.On Mar 19, 2024, at 4:14 PM, Mats Lidell ***@***.***> wrote:﻿
@matsl requested your review on: #492 Use numbers for line and column in links to org headers, add tests.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

matsl commented:
> Yes, there must be a number following the L and C or the link is invalid.

Sorry if I was not clear enough. Seems the link that is currently created using the master branch is with an L and C preceding the number for the line and column. However that link is is not valid on my machine at least. But if the L and C is dropped and only the numbers are used the links are valid!? Hence the question if this might be a bug? 

This PR changes the code so that such a link, without L and C, is created. There are test cases that verifies that the new links looks right.

I see now that the tests could also be improved to also verify that the links works as they should. I will look into that.

rswgnu commented:
The L and C are optional and the link should work with them; otherwise, it is a bug we should fix.On Mar 20, 2024, at 6:45 AM, Mats Lidell ***@***.***> wrote:﻿

Yes, there must be a number following the L and C or the link is invalid.

Sorry if I was not clear enough. Seems the link that is currently created using the master branch is with an L and C preceding the number for the line and column. However that link is is not valid on my machine at least. But if the L and C is dropped and only the numbers are used the links are valid!? Hence the question if this might be a bug?
This PR changes the code so that such a link, without L and C, is created. There are test cases that verifies that the new links looks right.
I see now that the tests could also be improved to also verify that the links works as they should. I will look into that.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

matsl commented:
> The L and C are optional and the link should work with them

Yes. On further testing today it seems to work. My bad. Sorry for the noise. Not sure why it broke yesterday. Probably bad manual testing on my part.

I can revert the removal of the optional letter (see comment below) and keep the tests since I believe they are an improvement.

But before that: Why do we add them here and not for the file paths links with line and column? Seems a bit add hoc to use it for the `string match` and not for the other path links? Should we not be consistent and use the same style everywhere for generated links?

matsl commented:
@rswgnu 

> I can revert the removal of the optional letter (see comment below) and keep the tests since I believe they are an improvement.

I have reverted the removal of L and C and updated the test consequently.

rswgnu commented:
> I have reverted the removal of L and C and updated the test consequently.

Further, we’ll want to unify the two link-directly types and have them both use the L and C.  I have added support for ‘I’ as an instance/occurrence number but have’t pushed that yet.  If you don’t get to this change today, I’ll probably make it tonight.

matsl commented:
> we’ll want to unify the two link-directly types and have them both use the L and C

The L and C was missing from `hpath:section-line-and-column-regexp`  which made `defib pathname-line-and-column` not identify pathnames with the L-C-format. 

This is in a sense a big change since, at least at a quick check, most docstrings related to the  pathname with line and column does not mention the optional L anc C characters. So before I go over that I just want to sync this change with you. If you think the change is good I'll complete the PR with the update to the docstrings and docs. (Alternatively we make that in to a separate PR?)

_There is something here that I have not fully got my head around this morning. I'm pretty sure the "string match" links using an L and C worked before although the link was not identified as a pathname. I tried to verify that running in a clean workspace but failed. Maybe I'm mixing up things so will pause this investigation in favor of your comments on the PR._

rswgnu commented:
As long as things work with both:  :L3, :3, :L3:C4, :3:4, then I am good with a fix.-- BobOn Mar 24, 2024, at 3:55 AM, Mats Lidell ***@***.***> wrote:﻿

we’ll want to unify the two link-directly types and have them both use the L and C

The L and C was missing from hpath:section-line-and-column-regexp  which made defib pathname-line-and-column not identify pathnames with the L-C-format.
This is in a sense a big change since, at least at a quick check, most docstrings related to the  pathname with line and column does not mention the optional L anc C characters. So before I go over that I just want to sync this change with you. If you think the change is good I'll complete the PR with the update to the docstrings and docs. (Alternatively we make that in to a separate PR?)
There is something here that I have not fully got my head around this morning. I'm pretty sure the "string match" links using an L and C worked before although the link was not identified as a pathname. I tried to verify that running in a clean workspace but failed. Maybe I'm mixing up things so will pause this investigation in favor of your comments on the PR.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

matsl commented:
> As long as things work with both:  :L3, :3, :L3:C4, :3:4, then I am good with a fix.

Well. It passes all tests. :shrug:  So I assume it means you see no apparent problem with the implemented solution. :smile: 

My next step is then to go over the docstrings so that the optional L and C are mentioned appropriately. I'll do that as part of this PR.

rswgnu commented on hui.el:
```diff
@@ -3,7 +3,7 @@
 ;;
 ;; Orig-Date:    19-Sep-91 at 21:42:03
-;; Last-Mod:     21-Mar-24 at 15:30:24 by Bob Weiner
+;; Last-Mod:     23-Mar-24 at 23:53:35 by Mats Lidell
```
Maybe you could just eliminate this timestamp change.

matsl replied:
Puh. Good catch. It is eliminated. (It took a bunch of commits due to my mistake but if we squash these it should be OK.)

Does this mean you are good with this PR, and I can do the docstrings update in another, or just that you wanted this removed when you spotted it was the only changed in that file?

rswgnu commented:
Youcan do the doc updates in another PR.
